utils.sh

```diff
@@ -94,58 +94,94 @@ function master_node_val() {
+          driver_prefix=ipmi
+          driver_interface=ipmitool
+      elif [ $driver == "idrac" ] ; then
+          driver=idrac
+          driver_prefix=drac
+          driver_interface=idrac
+      fi
+
+      name=$(master_node_val ${master_idx} "name")
+      mac=$(master_node_val ${master_idx} "ports[0].address")
+      cat <<EOF
+        openshift-master-$master_idx:
+          name: $name
+          port_address: "${mac}"
+          driver: "${driver}"
+          management_interface: "${driver_interface}"
+          power_interface: "${driver_interface}"
+          vendor_interface: "no-vendor"
+EOF
+
+    done
+
+    cat <<EOF
+      properties:
+EOF
+
+    for ((master_idx=0;master_idx<$1;master_idx++)); do
+      local_gb=$(master_node_val ${master_idx} "properties.local_gb")
+      cpu_arch=$(master_node_val ${master_idx} "properties.cpu_arch")
+
+      cat <<EOF
+        openshift-master-$master_idx:
+          local_gb: "${local_gb}"
+          cpu_arch: "${cpu_arch}"
+EOF
 
-    deploy_kernel=$(master_node_val ${master_idx} "driver_info.deploy_kernel")
-    deploy_ramdisk=$(master_node_val ${master_idx} "driver_info.deploy_ramdisk")
+    done
 
     cat <<EOF
-      master_$master_idx:
-        name: $name
-        port_address: "${mac}"
-        driver: "${driver}"
-        management_interface: "${driver_interface}"
-        power_interface: "${driver_interface}"
-        vendor_interface: "no-vendor"
-      properties_$master_idx:
-        local_gb: "${local_gb}"
-        cpu_arch: "${cpu_arch}"
-      root_device_$master_idx:
-        name: "${ROOT_DISK}"
-      driver_info_$master_idx:
-        ${port_prefix}${port_newline}${driver_prefix}_username: "${username}"
-        ${driver_prefix}_password: "${password}"
-        ${driver_prefix}_address: "${address}"
-        deploy_kernel:  "${deploy_kernel}"
-        deploy_ramdisk: "${deploy_ramdisk}"
+      root_devices:
+EOF
+
+    for ((master_idx=0;master_idx<$1;master_idx++)); do
+      cat <<EOF
+        openshift-master-$master_idx:
+          name: "${ROOT_DISK}"
 EOF
+    done
+
+    cat <<EOF
+      driver_infos:
+EOF
+
+    for ((master_idx=0;master_idx<$1;master_idx++)); do
+      port=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_port // \"\"")
+      if [ -n "$port" ]; then
+  	    port_prefix="${driver_prefix}_port: \"${port}\""
+        port_newline=$'\n        '
+      fi
+      username=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_username")
+      password=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_password")
+      address=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_address")
+
+      deploy_kernel=$(master_node_val ${master_idx} "driver_info.deploy_kernel")
+      deploy_ramdisk=$(master_node_val ${master_idx} "driver_info.deploy_ramdisk")
+
+      cat <<EOF
+        openshift-master-$master_idx:
+          ${port_prefix}${port_newline}${driver_prefix}_username: "${username}"
```

Comment:
With an IPMI port, the indentation is off a little on username (causes an error):

```
      driver_infos:
        openshift-master-0:
          ipmi_port: "6230"
        ipmi_username: "admin"
          ipmi_password: "password"
          ipmi_address: "192.168.111.1"
          deploy_kernel:  "http://172.22.0.1/images/ironic-python-agent.kernel"
          deploy_ramdisk: "http://172.22.0.1/images/ironic-python-agent.initramfs"
```

Error:

```
level=fatal msg="failed to fetch Master Machines: failed to load asset \"Install Config\": failed to unmarshal: error converting YAML to JSON: yaml: line 49: did not find expected key"
```